File: puzzles/python3/ascii-art/ascii_art.py
if __name__ == "__main__":
    width = int(input())
    height = int(input())
    text = input().upper()
    for _ in range(height):
        row = input()
        output = ""
        for character in text:
            position = ord(character) - ord('A')
            if position < 0 or position > 25:
                position = 26
            start = position * width
            end = start + width
            output += row[start:end]
        print(output)

File: puzzles/kotlin/src/ascii-art/ascii-art.kt
import java.util.*

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)
  val width = input.nextInt()
  val height = input.nextInt()
  if (input.hasNextLine()) {
    input.nextLine()
  }
  val text = input.nextLine().toUpperCase()

  for (i in 0 until height) {
    val row = input.nextLine()
    val output = StringBuilder()
    for (character in text) {
      var position = character.toInt() - 'A'.toInt()
      if (position < 0 || position > 25) {
        position = 26
      }
      val start = position * width
      val end = position * width + width
      output.append(row.substring(start, end))
    }
    println(output.toString())
  }
}
